dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAIChatCompletion_RequiredFunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,399 @@
+
+    //    Assert.Single(invokedFunctions);
+    //    Assert.Contains("GetCurrentDate", invokedFunctions);
+    //}
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorToInvokeKernelFunctionManuallyAsync()
+    {
+        // Arrange
+        this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+        var invokedFunctions = new List<string>();
+
+        this._autoFunctionInvocationFilter.RegisterFunctionInvocationHandler(async (context, next) =>
+        {
+            invokedFunctions.Add(context.Function.Name);
+            await next(context);
+        });
+
+        var settings = new OpenAIPromptExecutionSettings() { FunctionChoiceBehavior = FunctionChoiceBehavior.Required(autoInvoke: false) };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("How many days until Christmas?");
+
+        // Act
+        var result = await this._chatCompletionService.GetChatMessageContentAsync(chatHistory, settings, this._kernel);
+
+        // Assert
+        Assert.NotNull(result);
+
+        Assert.Empty(invokedFunctions);
+
+        var functionCalls = FunctionCallContent.GetFunctionCalls(result);
+        Assert.NotNull(functionCalls);
+        Assert.Single(functionCalls);
+
+        var functionCall = functionCalls.First();
+        Assert.Equal("DateTimeUtils", functionCall.PluginName);
+        Assert.Equal("GetCurrentDate", functionCall.FunctionName);
+    }
+
+    [Fact]
+    public async Task SpecifiedInCodeInstructsConnectorToInvokeKernelFunctionAutomaticallyForStreamingAsync()
+    {
+        // Arrange
+        this._kernel.ImportPluginFromType<DateTimeUtils>();
+
+        var invokedFunctions = new List<string?>();
+
+        IReadOnlyList<KernelFunction>? SelectFunctions(FunctionChoiceBehaviorFunctionsSelectorContext context)
+        {
+            // Get all function names that have been invoked
+            var invokedFunctionNames = context.ChatHistory
+                .SelectMany(m => m.Items.OfType<FunctionResultContent>())
+                .Select(i => i.FunctionName);
+
+            invokedFunctions.AddRange(invokedFunctionNames);
+
+            if (invokedFunctionNames.Contains("GetCurrentDate"))
+            {
+                return null; // Don't advertise any more functions because the expected function has been invoked.
+            }
+
+            return context.Functions;
+        }
+
+        var settings = new OpenAIPromptExecutionSettings { FunctionChoiceBehavior = FunctionChoiceBehavior.Required(autoInvoke: true, functionsSelector: SelectFunctions) };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddUserMessage("How many days until Christmas?");
+
+        string result = "";
+
+        // Act
+        await foreach (var content in this._chatCompletionService.GetStreamingChatMessageContentsAsync(chatHistory, settings, this._kernel))
+        {
+            result += content;
+        }
+
+        // Assert
+        Assert.NotNull(result);
```

Comment:
nit. `result` will never be null
```suggestion
        Assert.NotNull(result);
```

Reply:
fixed, thanks.